numpy/core/src/common/simd/intdiv.h

```diff
@@ -46,9 +46,6 @@
  *  - For 64-bit division on Aarch64 and IBM/Power, we fall-back to the scalar division
  *    since emulating multiply-high is expensive and both architectures have very fast dividers.
  *
- ** TODO:
- *   - Add support for Power10(VSX4)
```

Comment:
add support for VSX4 should include using intrinsic `vec_mulh` instead of `mulo, mule, and permute` within intrinsics `npyv_divc_##sfx` see:

https://github.com/numpy/numpy/blob/0e645362550c1d7f5a13077b5ecb7f4d9a60a167/numpy/core/src/common/simd/vsx/arithmetic.h#L108-L110

also uses `vec_div()` within `npyv_divc_u64`

https://github.com/numpy/numpy/blob/0e645362550c1d7f5a13077b5ecb7f4d9a60a167/numpy/core/src/common/simd/vsx/arithmetic.h#L214-L218

Reply:
After modifying the universal intrinsic `divc` following your suggestion, I see speedups between ~5-10%.

You will notice that I only modified the datatypes u32, s32, and u64. For the other data types, [su]8 and [su]16, as the POWER ISA 3.1 does not have `vmulh` for byte and half-word, the changes would require extra instructions (e.g., merge, pack, etc) that would lead to a slowdown of ~20-25%. Because of that, I left them without changes.